hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -209,15 +217,25 @@ protected BaseTableMetadata(HoodieEngineContext engineContext, String datasetBas
       // Validate the Metadata Table data by listing the partitions from the file system
       timer.startTimer();
 
-      // Ignore partition metadata file
-      HoodieTableMetaClient metaClient = new HoodieTableMetaClient(hadoopConf.get(), datasetBasePath);
-      FileStatus[] directStatuses = metaClient.getFs().listStatus(partitionPath,
-          p -> !p.getName().equals(HoodiePartitionMetadata.HOODIE_PARTITION_METAFILE));
-      metrics.ifPresent(m -> m.updateMetrics(HoodieMetadataMetrics.VALIDATE_FILES_STR, timer.endTimer()));
-
-      List<String> directFilenames = Arrays.stream(directStatuses)
-          .map(s -> s.getPath().getName()).sorted()
+      String partitionPathStr = FSUtils.getRelativePartitionPath(new Path(datasetMetaClient.getBasePath()), partitionPath);
+      String latestDataInstantTime = getLatestDatasetInstantTime();
+      HoodieTableFileSystemView dataFsView = new HoodieTableFileSystemView(datasetMetaClient, datasetMetaClient.getActiveTimeline());
```

Comment:
this validation stuff is actually tricky. we should do it once at the end of each metadata sync alone. If we turn this on mid stream during a commit, then the filesystem will report the newly written/inflight files as well, which the metdata table does not have and validation will always fail. this is kind of broken atm really.